src/hooks/swap/index.ts

```diff
@@ -68,17 +68,17 @@ export function useIsAmountPopulated() {
   return Boolean(useAtomValue(amountAtom))
 }
 
-export function useSwapAmount(field: Field): [string | undefined, (amount: string) => void] {
+export function useSwapAmount(field: Field): [string | undefined, (amount: string, origin?: 'max') => void] {
   const value = useAtomValue(amountAtom)
   const isFieldIndependent = useIsSwapFieldIndependent(field)
   const amount = isFieldIndependent ? value : undefined
 
   const { onAmountChange } = useAtomValue(swapEventHandlersAtom)
   const setSwap = useUpdateAtom(swapAtom)
   const updateAmount = useCallback(
-    (update: string) => {
+    (update: string, origin?: 'max') => {
```

Comment:
why the default here?

Reply:
not a default, a typing to only allow 'max'. I did not want a boolean so that it would be extensible in the future.